DQM/TrackerMonitorTrack/src/MonitorTrackResiduals.cc

```diff
@@ -229,28 +222,41 @@ void MonitorTrackResiduals::analyze(const edm::Event& iEvent, const edm::EventSe
   iSetup.get<TrackerTopologyRcd>().get(tTopoHandle);
   const TrackerTopology* const tTopo = tTopoHandle.product();
 
-  std::vector<TrackerValidationVariables::AVHitStruct> v_hitstruct;
-  avalidator_.fillHitQuantities(iEvent,v_hitstruct);
-  for (auto it : v_hitstruct) {
-    uint RawId = it.rawDetId;
-
-    if (ModOn) {
-      auto& mod_histos = m_ModuleResiduals[std::make_pair("",RawId)];
-      mod_histos.x.base->Fill(it.resXprime);
-      mod_histos.x.normed->Fill(it.resXprime/it.resXprimeErr);
-      mod_histos.y.base->Fill(it.resYprime);
-      mod_histos.y.normed->Fill(it.resYprime/it.resYprimeErr);
-    }
-    auto subdetandlayer = findSubdetAndLayer(RawId, tTopo);
-    auto histos = m_SubdetLayerResiduals[subdetandlayer];
-    // fill if its error is not zero
-    if(it.resXprimeErr != 0 && histos.x.base) {
-      histos.x.base->Fill(it.resXprime);
-      histos.x.normed->Fill(it.resXprime/it.resXprimeErr);
-    }
-    if(it.resYprimeErr != 0 && histos.y.base) {
-      histos.y.base->Fill(it.resYprime);
-      histos.y.normed->Fill(it.resYprime/it.resYprimeErr);
+  avalidator_.fillTrackQuantities(iEvent, iSetup, 
+                  // tell the validator to only look at good tracks
+                  [](const reco::Track& track) -> bool { 
+                    return track.pt() > 0.75
+                        && abs( track.dxy() ) > 5*track.dxyError();
```

Comment:
Il 07/03/2016 16.54, Marco Musich ha scritto:
>
> In DQM/TrackerMonitorTrack/src/MonitorTrackResiduals.cc 
> <https://github.com/cms-sw/cmssw/pull/13393#discussion_r55222938>:
>
> > -    }
> > -    auto subdetandlayer = findSubdetAndLayer(RawId, tTopo);
> > -    auto histos = m_SubdetLayerResiduals[subdetandlayer];
> > -    // fill if its error is not zero
> > -    if(it.resXprimeErr != 0 && histos.x.base) {
> > -      histos.x.base->Fill(it.resXprime);
> > -      histos.x.normed->Fill(it.resXprime/it.resXprimeErr);
> > -    }
> > -    if(it.resYprimeErr != 0 && histos.y.base) {
> > -      histos.y.base->Fill(it.resYprime);
> > -      histos.y.normed->Fill(it.resYprime/it.resYprimeErr);
> > +  avalidator_.fillTrackQuantities(iEvent, iSetup,
> > +                  // tell the validator to only look at good tracks
> > +                  [](const reco::Track& track) -> bool {
> > +                    return track.pt() > 0.75
> > +                        && abs( track.dxy() ) > 5*track.dxyError();
>
> @fioriNTU <https://github.com/fioriNTU> can you please comment? I feel 
> like you are going to badly bias your selection here....
>
There is an error in this cut, in the original code it was

if(abs( track.dxy() ) > 5*track.dxyError()) continue;

now the logic is inverted. Of course we don't want to select displaced tracks. Marcel update the PR, thanks for spotting!

> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/cms-sw/cmssw/pull/13393/files#r55222938>.
>



---
Questa e-mail è stata controllata per individuare virus con Avast antivirus.
https://www.avast.com/antivirus